src/sas/qtgui/Utilities/TabbedModelEditor.py

```diff
@@ -432,24 +425,47 @@ def updateFromEditor(self):
         if not w.is_python:
             base, _ = os.path.splitext(filename)
             filename = base + '.c'
-
         # make sure we have the file handle ready
-        assert(filename != "")
+        assert filename != ""
+
         # Retrieve model string
         model_str = self.getModel()['text']
+        # Save the file
+        self.writeFile(filename, model_str)
+
+        # Get model filepath
+        plugin_location = models.find_plugins_dir()
+        full_path = os.path.join(plugin_location, filename)
+        if not w.is_python and self.is_python:
+            pass
+        elif os.path.splitext(full_path)[1] != ".py":
```

Comment:
No idea where `filename` is coming from, but if it is `model.c` and the code is python would you want this saved as `model.c.py`? This will likely cause problems for the loader. 

You ought to check that filename without extension is a valid python symbol name [with something like `Path(filename).stem.isidentifier()`] and replace the extension with `.py` for python models.

Shouldn't this happen before `self.writeFile()` ?

Also, `self.writeFile()` must have similar code for path handling. Can you modify that function to return the full path or set it as an attribute?